common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -465,7 +470,11 @@ public static void copyMemory(long srcAddr, byte[] dst, int dstIndex, long lengt
      * consumer (one thread!).
      */
     public static <T> Queue<T> newMpscQueue() {
-        return new MpscLinkedQueue<T>();
+        if (hasUnsafe()) {
+            return JAVA_VERSION < 8 ? new MpscLinkedQueue7<T>() : new MpscLinkedQueue8<T>();
```

Comment:
if Netty 4 is for JDK 5 I think you need to test fully this branch with JDK 5, Unsafe won't be the problem but JCtools source code may be.